src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -0,0 +1,607 @@
+    def __init__(
+        self, name: SessionsQueryFunction, column: _VirtualColumnName, group_by_status: bool
+    ) -> None:
+        self._name = name
+        self._column = column
+
+        self.group_by_status = group_by_status
+
+    def get_name(self) -> SessionsQueryFunction:
+        return self._name
+
+    def get_values(self, data_points: _DataPoints, key: _DataPointKey) -> Sequence[_StatusValue]:
+        assert key.raw_session_status is None, key  # session.duration does not have status tag
+        value = 1000.0 * data_points[key]  # sessions backend stores milliseconds
+        if self.group_by_status:
+            # Only 'healthy' sessions have duration data:
+            return [
+                _StatusValue("abnormal", None),
+                _StatusValue("crashed", None),
+                _StatusValue("errored", None),
+                _StatusValue("healthy", value),
+            ]
+        else:
+            return [
+                _StatusValue(None, value),
+            ]
+
+
+def _fetch_data(
+    org_id: int,
+    query: QueryDefinition,
+) -> Tuple[_SnubaDataByMetric, Mapping[Tuple[_MetricName, _VirtualColumnName], _OutputField]]:
+    """Build & run necessary snuba queries"""
+    conditions = [
+        Condition(Column("org_id"), Op.EQ, org_id),
+        Condition(Column("project_id"), Op.IN, query.filter_keys["project_id"]),
+        Condition(Column(TS_COL_QUERY), Op.GTE, query.start),
+        Condition(Column(TS_COL_QUERY), Op.LT, query.end),
+    ]
+
+    conditions.extend(_get_filter_conditions(org_id, query.conditions))
+
+    # It greatly simplifies code if we just assume that these two tags exist:
+    # TODO: Can we get away with that assumption?
+    tag_key_session_status = _resolve_ensured(org_id, "session.status")
+
+    groupby_tags = [field for field in query.raw_groupby if field != "project"]
+
+    tag_keys = {field: _resolve(org_id, field) for field in groupby_tags}
+    groupby = {
+        field: Column(f"tags[{tag_id}]")
+        for field, tag_id in tag_keys.items()
+        if tag_id is not None  # exclude unresolved keys from groupby
+    }
+
+    if "project" in query.raw_groupby:
+        groupby["project"] = Column("project_id")
+
+    data: List[Tuple[_MetricName, _SnubaData]] = []
+
+    #: Find the field that needs a specific column in a specific metric
+    metric_to_output_field: MutableMapping[
+        Tuple[_MetricName, _VirtualColumnName], _OutputField
+    ] = {}
+
+    def get_query(
+        entity_key: EntityKey,
+        metric_id: int,
+        columns: Sequence[str],
+        series: bool,
+        extra_conditions: List[Condition],
+        remove_groupby: Set[Column],
+    ) -> Query:
+        """Build the snuba query"""
+        full_groupby = set(groupby.values()) - remove_groupby
+        if series:
+            full_groupby.add(Column(TS_COL_GROUP))
+
+        where = conditions + [Condition(Column("metric_id"), Op.EQ, metric_id)] + extra_conditions
```

Comment:
Nit: 
```
conditions += extra_conditions + [Condition(Column("metric_id"), Op.EQ, metric_id)]
```

Reply:
`conditions` is defined outside of this function, so we shouldn't mutate it. I'll rename it to `base_conditions` though.